### PR TITLE
Deltaproxy concurrent

### LINKED ISSUE #61153: [FEATURE REQUEST] Parallelize initial connections to devices when using Deltaproxy
**Is your feature request related to a problem? Please describe.**
Currently when starting the deltaproxy process, it connects to each device in sequence, waiting for commands to be executed on the device, before starting the connection to any other devices. When connecting to slower network devices, even with a modest number of devices, this can cause process startup to take many 10's of minutes to complete initialization and be ready to accept commands.

**Describe the solution you'd like**
Initialize connections in parallel (perhaps make it the number of parallel connections configurable), to speed up initial connection setup.

**Describe alternatives you've considered**
Run proxy minions instead at a cost of huge amounts of memory, or run more deltaproxy processes with fewer devices assigned per-process.

**Additional context**
Previously discussed with @garethgreenaway [here](https://github.com/saltstack/salt/issues/60177#issuecomment-841544376).

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Deltaproxy concurrent experiment

### What issues does this PR fix or reference?
Fixes: #61153

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by s0undt3ch — COMMENTED** [review]
I like where this is going.

File: changelog/61153.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Initial work to allow parallel startup of proxy minions when used as sub proxies with Deltaproxy.
```
Comment (Ch3LL) [inline changelog/61153.fixed:1]: File extension should be `.added` since this is a new feature.

File: salt/metaproxy/deltaproxy.py [hunk 2]
```diff
@@ -320,168 +320,224 @@ def post_master_init(self, master):
...
-        control_id = self.opts["id"]
-        proxyopts = self.opts.copy()
-        proxyopts["id"] = _id
 
-        proxyopts = salt.config.proxy_config(
-            self.opts["conf_file"], defaults=proxyopts, minion_id=_id
-        )
-        proxyopts["id"] = proxyopts["proxyid"] = _id
+    if self.opts["proxy"].get("parallel_startup"):
+        log.debug("Doing parallel startup")
```
Comment (Ch3LL) [inline salt/metaproxy/deltaproxy.py:325]:
```suggestion
        log.debug("Initiating parallel startup for proxies")
```

File: salt/metaproxy/deltaproxy.py [hunk 3]
```diff
@@ -320,168 +320,224 @@ def post_master_init(self, master):
...
+                self.deltaproxy_objs[minion_id] = sub_proxy_data["proxy_minion"]
+
+                if self.deltaproxy_opts[minion_id] and self.deltaproxy_objs[minion_id]:
+                    self.deltaproxy_objs[
+                        minion_id
+                    ].req_channel = salt.transport.client.AsyncReqChannel.factory(
+                        sub_proxy_data["proxy_opts"], io_loop=self.io_loop
+                    )
+    else:
+        log.debug("Doing non-parallel startup")
```
Comment (Ch3LL) [inline salt/metaproxy/deltaproxy.py:354]:
```suggestion
        log.debug("Initiating non-parallel startup for proxies")
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**